### PR TITLE
fix(api): Only track attributes with valid attribute types.

### PR DESCRIPTION
**Pull request by mfahadahmed**

### CODE REVIEW

**Comment by msohailhussain** [comment]
I need to review it, will do it later today.

File: OptimizelySDK/Utils/Validator.cs [hunk 1]
```diff
@@ -103,14 +103,24 @@ public static bool IsFeatureFlagValid(ProjectConfig projectConfig, FeatureFlag f
             return true;
         }
 
+        private static Type[] AttributeTypes =
```
Comment (msohailhussain) [inline OptimizelySDK/Utils/Validator.cs]: typeof generally takes too much time to get the type. Why we just don't replace with attribute types with string[].

File: OptimizelySDK/Utils/Validator.cs [hunk 2]
```diff
@@ -103,14 +103,24 @@ public static bool IsFeatureFlagValid(ProjectConfig projectConfig, FeatureFlag f
...
+
         /// <summary>
         /// Determine if given user attribute is valid.
         /// </summary>
         /// <param name="attribute">Attribute key and value pair</param>
         /// <returns>true if attribute key is not null and value is one of the supported type, false otherwise</returns>
         public static bool IsUserAttributeValid(KeyValuePair<string, object> attribute)
         {
-            return !string.IsNullOrEmpty(attribute.Key) && attribute.Value is string || attribute.Value is bool || attribute.Value is int || attribute.Value is double;
+            return !string.IsNullOrEmpty(attribute.Key) && AttributeTypes.Contains(attribute.Value?.GetType());
```
Comment (msohailhussain) [inline OptimizelySDK/Utils/Validator.cs]: .GetType().Name returns string, which we can use.

**Comment by msohailhussain** [comment]
@mikeng13 Thanks Mike, once Python PR is merged, I will take a look at this PR.

File: OptimizelySDK.Tests/UtilsTests/ValidatorTest.cs [hunk 3]
```diff
@@ -136,5 +137,46 @@ public void TestAreEventTagsValidInvalidTags()
...
+            };
+
+            foreach (var attribute in userAttributes)
+                Assert.True(Validator.IsUserAttributeValid(attribute));
+        }
+
+        [Test]
+        public void TestIsUserAttributeValidWithInvalidValues()
+        {
+            var userAttributesInvalid = new UserAttributes
```
Comment (mikeng13) [inline OptimizelySDK.Tests/UtilsTests/ValidatorTest.cs]: nit: Can we call this `invalidUserAttributes`

**Pull request merged**